src/main/java/co/omise/models/Source.java

```diff
@@ -347,6 +360,9 @@ public CreateRequestBuilder platformType(PlatformType platformType) {
 
         @Override
         protected RequestBody payload() throws IOException {
+            if (type == SourceType.WechatPay && (ip == null || ip.isEmpty())) {
```

Comment:
Maybe we should add some commentary to justify why we are doing this validation on the client-side: the server can't know whether the caller is coming from a customer device or not.

Reply:
This `payload()` is shared by all the APMs. Can we move this validation logic to a different class or function for now? This sets the groundwork for handling potential unique requirements in different APMs without cluttering the main code with numerous if statements.

Reply:
The check has been removed.